nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/test/java/org/nuxeo/ecm/automation/core/operations/services/TestWorkManagerRunWorkInFailure.java

```diff
@@ -0,0 +1,120 @@
+ * limitations under the License.
+ *
+ * Contributors:
+ *     bdelbosc
+ */
+package org.nuxeo.ecm.automation.core.operations.services;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.automation.AutomationService;
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.automation.core.operations.services.workmanager.WorkManagerRunWorkInFailure;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.impl.blob.JSONBlob;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.core.test.annotations.Granularity;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.ecm.core.work.SleepWork;
+import org.nuxeo.ecm.core.work.api.Work;
+import org.nuxeo.ecm.core.work.api.WorkManager;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.stream.RuntimeStreamFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ RuntimeStreamFeature.class, CoreFeature.class })
+@Deploy("org.nuxeo.ecm.automation.core")
+@Deploy("org.nuxeo.ecm.automation.features")
+@Deploy("org.nuxeo.ecm.core.event:test-work-dead-letter-queue.xml")
+@RepositoryConfig(cleanup = Granularity.METHOD)
+public class TestWorkManagerRunWorkInFailure {
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected AutomationService service;
+
+    protected OperationContext ctx;
+
+    @Before
+    public void createOperationContext() {
+        ctx = new OperationContext(session);
+    }
+
+    @After
+    public void closeOperationContext() {
+        ctx.close();
+    }
+
+    @Test
+    public void testWorkManagerRunWorkInFailureOperation() throws InterruptedException, OperationException {
+        // Submit a work that fails and goes to the dead letter queue
+        Work failWork = new FailingWork(10);
+        WorkManager wm = Framework.getService(WorkManager.class);
+        wm.schedule(failWork);
+        assertTrue(wm.awaitCompletion(2000, TimeUnit.MILLISECONDS));
+
+        // Run the operation to replay the work in failure
+        Map<String, Serializable> params = new HashMap<>();
+
+        JSONBlob result = (JSONBlob) service.run(ctx, WorkManagerRunWorkInFailure.ID, params);
+        assertNotNull(result);
+        // One work re-processed without success
+        assertEquals("[{\"total\":1,\"success\":0}]", result.getString());
```

Comment:
The JSON comes from a `HashMap`, order is not guaranteed. You should do JSON-level checks.